overlord/hookstate/hookmgr.go

```diff
@@ -295,6 +318,19 @@ func runHookAndWait(snapName string, revision snap.Revision, hookName, hookConte
 		return nil, err
 	}
 
+	// add timeout handling
+	var killTimer *time.Timer
+	if maxRuntime > 0 {
+		killTimer = time.AfterFunc(maxRuntime, func() {
+			if err := killemAll(command); err != nil {
+				logger.Noticef("cannot kill %q: %s", hookName, err)
+			}
+			// ensure we log what happened
+			fmt.Fprintf(buffer, "\nexceeded maximum runtime of %s\n", maxRuntime)
```

Comment:
Indeed, I removed this now in favour of using the appropriate error message in the killTimeCh case.